Missing hip_add_library in cmake

## Issue

We are trying to use `hip_add_library` in cmake with chipStar. It works ok with Rocm 6.3.2 (see reproducer below). However when we try with chipStar 1.2.1 we get the following error:

```
CMake Error at CMakeLists.txt:12 (HIP_ADD_LIBRARY):
  Unknown CMake command "HIP_ADD_LIBRARY".
```

## Reproducer and expected result:

```
> cat main.cc 
#include <iostream>

void printHipVersion();

int main()
{
    std::cout << "Hello, world!" << std::endl;

    std::cout << "CUDA: On" << std::endl;
    printHipVersion();

    return 0;
}

> cat gpu.cpp 
#include <iostream>
#include <hip/hip_runtime.h>

void printHipVersion()
{
//    std::cout << "CUDA Compiled version: " << __CUDACC_VER__ << std::endl;

    int runtime_ver;
    hipRuntimeGetVersion(&runtime_ver);
    std::cout << "CUDA Runtime version: " << runtime_ver << std::endl;

    int driver_ver;
    hipDriverGetVersion(&driver_ver);
    std::cout << "CUDA Driver version: " << driver_ver << std::endl;
}
```
CMakeLists.txt:
```
> cat CMakeLists.txt 
cmake_minimum_required(VERSION 3.5.1)
message(STATUS "CMake version: ${CMAKE_VERSION}")

project(diana)

list(APPEND CMAKE_MODULES_PREFIX $ENV{ROCM_PATH}/cmake)
list(APPEND CMAKE_MODULE_PATH $ENV{ROCM_PATH}/lib/cmake/hip/)
list(APPEND CMAKE_PREFIX_PATH $ENV{ROCM_PATH})

add_executable(diana main.cc)
find_package(HIP REQUIRED)
HIP_ADD_LIBRARY(diana_gpu gpu.cpp)
target_link_libraries(diana diana_gpu)
```

Building with Rocm 6.3.2 (successful):
```
> mkdir build
> cd build
> cmake -DCMAKE_CXX_COMPILER=hipcc ../
-- CMake version: 3.28.3
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is Clang 18.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /soft/compilers/rocm/rocm-6.3.2/bin/hipcc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found HIP: /soft/compilers/rocm/rocm-6.3.2 (found version "6.3.42134-a9a80e791")
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS - Success
-- Configuring done (3.9s)
-- Generating done (0.0s)
-- Build files have been written to: /home/bertoni/hip-training/simple/cmake-cuda-example/build
(base) bertoni@amdgpu04:~/hip-training/simple/cmake-cuda-example/build> make
[ 25%] Building CXX object CMakeFiles/diana_gpu.dir/gpu.cpp.o
/home/bertoni/hip-training/simple/cmake-cuda-example/gpu.cpp:9:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
    9 |     hipRuntimeGetVersion(&runtime_ver);
      |     ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~
/home/bertoni/hip-training/simple/cmake-cuda-example/gpu.cpp:13:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
   13 |     hipDriverGetVersion(&driver_ver);
      |     ^~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~
2 warnings generated when compiling for gfx90a.
/home/bertoni/hip-training/simple/cmake-cuda-example/gpu.cpp:9:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
    9 |     hipRuntimeGetVersion(&runtime_ver);
      |     ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~
/home/bertoni/hip-training/simple/cmake-cuda-example/gpu.cpp:13:5: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
   13 |     hipDriverGetVersion(&driver_ver);
      |     ^~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~
2 warnings generated when compiling for host.
[ 50%] Linking CXX static library libdiana_gpu.a
[ 50%] Built target diana_gpu
[ 75%] Building CXX object CMakeFiles/diana.dir/main.cc.o
[100%] Linking CXX executable diana
[100%] Built target diana

```

## Workaround

If we pass in our own FindHIP.cmake file (as with `cmake -DCMAKE_CXX_COMPILER=hipcc -DCMAKE_MODULE_PATH=/home/bertoni/hip/cmake/ ../`) it will compile.

Thanks!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -636,6 +636,19 @@ set(CPACK_PROJECT_NAME ${PROJECT_NAME})
 set(CPACK_PROJECT_VERSION ${PROJECT_VERSION})
 include(CPack)
 
+# Install FindHIP.cmake and related files
+install(FILES 
+  ${CMAKE_SOURCE_DIR}/HIP/cmake/FindHIP.cmake
+  DESTINATION ${CONFIG_PACKAGE_INSTALL_DIR}
+)
+
+# Install FindHIP helper files in the correct subdirectory
+install(FILES 
+  ${CMAKE_SOURCE_DIR}/HIP/cmake/FindHIP/run_hipcc.cmake
+  ${CMAKE_SOURCE_DIR}/HIP/cmake/FindHIP/run_make2cmake.cmake
+  DESTINATION ${CONFIG_PACKAGE_INSTALL_DIR}/FindHIP
+)
+
 # Setup .hipVersion
 # Install .hipVersion for hipcc. The file is used to detect a HIP
 # runtime and clang extracts HIP version from it.

